android/modules/media/src/java/ti/modules/titanium/media/MediaModule.java

```diff
@@ -1155,13 +1155,15 @@ public void previewImage(KrollDict options)
 		TiActivitySupport activitySupport = (TiActivitySupport) activity;
 
 		Intent intent = new Intent(Intent.ACTION_VIEW);
+		intent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
 		TiIntentWrapper previewIntent = new TiIntentWrapper(intent);
 		String mimeType = image.getMimeType();
+		Uri imageUri = TiFileProvider.createUriFrom(f.getNativeFile());
```

Comment:
The `getNativeFile()` method can return `null` if:
- The photo came from the cloud.
- The image was loaded from a blob. (Can test with `View.toImage()`.)

Calling `getNativeFile()` will crash with a `NullPointerException` for the above case too unless my code changes from PR #9224 gets merged in. (Required fixes to `TitaniumBlob` and `TiBlob`.)

Ideally, we should store the image to a temp file if it's in pure blob form before previewing it.

Reply:
We can make this a separate bug-fix/PR if you want, because it looks like the issue I mentioned was always there.

Reply:
Yeah, this should be a separate issue

Reply:
Okay. I just wrote up a ticket with a reproducible case here...
https://jira.appcelerator.org/browse/TIMOB-25619